src/write_csv.jl

```diff
@@ -15,12 +15,18 @@ function write_csv(df::DataFrame, erupt_saved::EruptSaved{Float64}, path::String
     ]
     rename!(df, ["timestamp" => "time"])
     rename!(df, ["value$i" => names[i] for i in 1:10])
+    # add eps_x column
+    m_h2o = df[:, "total_mass_H2O"] ./ df[:, "total_mass"]
+    m_co2 = df[:, "total_mass_CO2"] ./ df[:, "total_mass"]
+    crystal_fraction_eps_x′(T, P, mH2O, mCO2) = crystal_fraction_eps_x(composition, T, P, mH2O, mCO2)
+    df[!, "eps_x"] = crystal_fraction_eps_x′.(df[:, "T"], df[:, "P+dP"], m_h2o, m_co2)
+
+    n = length(erupt_saved.time)
     df_erupt = DataFrame(
-        reduce(
-            hcat,
-            [erupt_saved.time, erupt_saved.duration, erupt_saved.mass, erupt_saved.volume],
-        ),
-        ["time_of_eruption", "duration_of_eruption", "mass_erupted", "volume_erupted"],
+        time_of_eruption = erupt_saved.time,
+        duration_of_eruption = [erupt_saved.duration; fill(NaN, n - length(erupt_saved.duration))],
+        mass_erupted = [erupt_saved.mass; fill(NaN, n - length(erupt_saved.mass))],
+        volume_erupted = [erupt_saved.volume; fill(NaN, n - length(erupt_saved.volume))]
```

Comment:
Here it would be sensible to create a function that does `[arr, fill(NaN, n - length(arr))]` and use it three times.